src/server/main_service.cc

```diff
@@ -882,7 +882,13 @@ void Service::DispatchCommand(CmdArgList args, facade::ConnectionContext* cntx)
     dfly_cntx->reply_builder()->CloseConnection();
   }
 
-  end_usec = ProactorBase::GetMonotonicTimeNs();
+  end_usec = ProactorBase::GetMonotonicTimeNs();  // misleading name, this is actually ns
+  auto after = absl::GetCurrentTimeNanos();
+
+  auto& ent = etl.cmd_stats_map[cid->name().data()];
+  ++ent.first;
+  ent.second += (after - before) / 1000;
+
```

Comment:
Why do we do the duration calculation twice?

Reply:
I think the ProactorBase one might be caching the time for performance reasons? So it kind of seems like the resolution loss is intended here. If the overhead of `GetMonotonicTimeNs` is acceptable here let's use it for both calculations

Reply:
hey @royjacobson  lets leave it as is. it's on me to check whether GetMonotonicTimeNs really works

Reply:
Sorry, I meant let's use `GetCurrentTimeNanos` for both calculations.
sounds good